referenceqvm/tests/test_unitary_generator.py

```diff
@@ -208,3 +210,36 @@ def test_tensor_gates_two_qubit():
     test_unitary = tensor_gates(gate_matrix, {}, prog.actions[0][1], 4).toarray()
     true_unitary = apply_gate(gate_matrix['CNOT'], [1, 3], 4).toarray()
     assert np.allclose(test_unitary, true_unitary)
+
+
+def test_tensor_up():
```

Comment:
this test has a lot of cases. You should make this a bit more atomic. Maybe split it up in the error testing cases and the correctness cases

Reply:
👍